### PR TITLE
feat: reduce Realtime footprint to four helper methods

### PR DESCRIPTION
**Pull request by w3b6x9**
## What kind of change does this PR introduce?

feature

## What is the new behavior?

Minimize Realtime and included Realtime-related methods in `SupabaseClient` are wrappers around `realtime-js` methods.

### CODE REVIEW

File: example/next-storage/components/ProfileList.tsx [hunk 1]
```diff
@@ -14,9 +13,9 @@ export default function ProfileList() {
     const realtimeProfiles = supabase
       .channel('profiles-channel')
       .on(
-        'realtime',
+        'postgres_changes',
         { event: '*', schema: 'public', table: 'profiles' },
-        (payload: SupabaseRealtimePayload<Profile>) => profileUpdated(profiles, payload.new)
+        (payload: { [key: string]: any }) => profileUpdated(profiles, payload.new)
```
Comment (w3b6x9) [inline example/next-storage/components/ProfileList.tsx:18]: will introduce a type for this

File: src/SupabaseClient.ts [hunk 2]
```diff
@@ -174,50 +168,31 @@ export default class SupabaseClient<
...
-    const allChanPromises = allChans.map((chan) => this.removeChannel(chan))
-    const allRemovedChans = await Promise.all(allChanPromises)
-
-    return allRemovedChans.map(({ error }, i) => {
-      return {
-        data: { channels: allChans[i] },
-        error,
-      }
-    })
+  removeChannel(channel: RealtimeChannel): Promise<any> {
```
Comment (w3b6x9) [inline src/SupabaseClient.ts]: will update return type

File: src/SupabaseClient.ts [hunk 3]
```diff
@@ -174,50 +168,31 @@ export default class SupabaseClient<
...
-    channel: RealtimeChannel
-  ): Promise<{ data: { openChannels: number }; error: Error | null }> {
-    const { error } = await this._closeChannel(channel)
-    const allChans: RealtimeChannel[] = this.getChannels()
-    const openChanCount = allChans.filter((chan) => chan.isJoined()).length
-
-    if (allChans.length === 0) await this.realtime.disconnect()
-
-    return { data: { openChannels: openChanCount }, error }
+  removeAllChannels(): Promise<any> {
```
Comment (w3b6x9) [inline src/SupabaseClient.ts]: will update return type

File: example/next-storage/components/ProfileList.tsx (hunk 1 above)
Comment (w3b6x9) [reply example/next-storage/components/ProfileList.tsx:18]: will do this later

**Review by inian — APPROVED** [review]

**Pull request merged**